File: config/plugins/gemini-master-suite/scripts/seo-folder-scanner.js
const fs = require('fs');
const path = require('path');

const targetDir = process.argv[2];

if (!targetDir) {
    console.log("Kullanım: node seo-folder-scanner.js <klasör_yolu>");
    process.exit(1);
}

function scanDirectory(dir) {
    let results = [];
    const files = fs.readdirSync(dir);
    
    for (const file of files) {
        const fullPath = path.join(dir, file);
        const stat = fs.statSync(fullPath);
        
        if (stat.isDirectory()) {
            results = results.concat(scanDirectory(fullPath));
        } else if (fullPath.endsWith('.html') || fullPath.endsWith('.htm')) {
            results.push(analyzeFile(fullPath, stat.size));
        }
    }
    return results;
}

function analyzeFile(filePath, byteSize) {
    const content = fs.readFileSync(filePath, 'utf8');
    
    // Vektör 1: Core Technical & Structural SEO
    const h1Count = (content.match(/<h1[\s>]/gi) || []).length;
    const hasCanonical = /<link[^>]*rel=["']canonical["'][^>]*>/i.test(content);
    const hasNoIndex = /<meta[^>]+name=["']robots["'][^>]+content=["'][^"']*noindex/i.test(content);
    
    // Vektör 4: LLMO (Large Language Model Optimization)
    const hasLlmsDiscovery = /rel=["']describedby["'][^>]+href=["']\/llms\.txt["']/i.test(content);
    const hasMarkdownAlternate = /rel=["']alternate["'][^>]+type=["']text\/markdown["']/i.test(content);
    
    // Vektör 6: RAG (Retrieval-Augmented Generation) & Dense Retrieval
    const hasDataChunkId = /data-chunk-id/i.test(content);
    const h2h3Count = (content.match(/<(h2|h3)[\s>]/gi) || []).length;
    
    // Vektör 7: E-E-A-T & Knowledge Vault
    const hasWikidata = /wikidata\.org\/wiki\/Q/i.test(content);
    
    return {
        file: filePath,
        size: byteSize,
        h1: h1Count,
        canonical: hasCanonical,
        noindex: hasNoIndex,
        llmsDiscovery: hasLlmsDiscovery,
        markdownAlt: hasMarkdownAlternate,
        dataChunkId: hasDataChunkId,
        h2h3: h2h3Count,
        wikidata: hasWikidata
    };
}

console.log(`\n=== SÜPER SEO MOTORU V3.0 (KAPSAMLI KLASÖR TARAMASI) ===`);
console.log(`[+] Hedef Dizin: ${path.resolve(targetDir)}\n`);

try {
    const report = scanDirectory(targetDir);
    
    if (report.length === 0) {
        console.log("[!] Belirtilen klasörde HTML dosyası bulunamadı.");
        process.exit(0);
    }
    
    report.forEach(res => {
        console.log(`📄 Dosya: ${res.file}`);
        
        // Vektör 2: 14KB Sınırı
        if (res.size > 14336) {
            console.log(`   [X] HATA (TOKEN-BLOAT-001): Boyut ${res.size} byte (14KB aşıldı!)`);
        } else {
            console.log(`   [V] PASS: Boyut ${res.size} byte (Uygun)`);
        }
        
        // Vektör 1: H1 ve Canonical
        console.log(res.h1 === 1 ? `   [V] PASS: Tek H1 etiketi (TECH-H1-001)` : `   [X] HATA: H1 Sayısı ${res.h1} (Sadece 1 adet olmalı)`);
        console.log(res.canonical ? `   [V] PASS: Canonical var (TECH-CANON-001)` : `   [X] HATA: Canonical YOK`);
        if (res.noindex) console.log(`   [X] KRİTİK HATA (TECH-NOINDEX-001): Accidental noindex tespit edildi!`);
        
        // Vektör 4: LLMO
        console.log(res.llmsDiscovery ? `   [V] PASS: llms.txt discovery (LLMS-DISCOVERY-001)` : `   [!] UYARI: describedby=llms.txt YOK`);
        
        // Vektör 6: RAG
        console.log(res.dataChunkId ? `   [V] PASS: Semantic boundaries data-chunk-id (RAG-CHUNK-001)` : `   [!] UYARI: data-chunk-id bulunamadı (RAG kopmaları riski)`);
        console.log(res.h2h3 >= 3 ? `   [V] PASS: H2/H3 Matrisi (${res.h2h3} adet) (COLBERT-MAXSIM-001)` : `   [!] UYARI: H2/H3 yapısı zayıf (${res.h2h3} adet)`);
        
        // Vektör 7: Knowledge Vault
        console.log(res.wikidata ? `   [V] PASS: Wikidata JSON-LD/sameAs referansı (ENTITY-VAULT-001)` : `   [!] UYARI: Wikidata (QID) entity referansı eksik`);
        
        console.log("--------------------------------------------------");
    });
    
} catch (err) {
    console.error(`[X] HATA: ${err.message}`);
}

File: mac/codex/codex-master-suite/scripts/python/seo_folder_scanner.py
import os
import sys
import re

def analyze_file(file_path):
    byte_size = os.path.getsize(file_path)
    
    with open(file_path, 'r', encoding='utf-8', errors='ignore') as f:
        content = f.read()
        
    # Vektör 1
    h1_count = len(re.findall(r'(?i)<h1[\s>]', content))
    has_canonical = bool(re.search(r'(?i)<link[^>]*rel=["\']canonical["\'][^>]*>', content))
    has_noindex = bool(re.search(r'(?i)<meta[^>]+name=["\']robots["\'][^>]+content=["\'][^"\']*noindex', content))
    
    # Vektör 4
    has_llms_discovery = bool(re.search(r'(?i)rel=["\']describedby["\'][^>]+href=["\']/llms\.txt["\']', content))
    has_markdown_alt = bool(re.search(r'(?i)rel=["\']alternate["\'][^>]+type=["\']text/markdown["\']', content))
    
    # Vektör 6
    has_data_chunk_id = bool(re.search(r'(?i)data-chunk-id', content))
    h2h3_count = len(re.findall(r'(?i)<(h2|h3)[\s>]', content))
    
    # Vektör 7
    has_wikidata = bool(re.search(r'(?i)wikidata\.org/wiki/Q', content))
    
    return {
        'file': file_path,
        'size': byte_size,
        'h1': h1_count,
        'canonical': has_canonical,
        'noindex': has_noindex,
        'llms_discovery': has_llms_discovery,
        'data_chunk_id': has_data_chunk_id,
        'h2h3': h2h3_count,
        'wikidata': has_wikidata
    }

def scan_directory(target_dir):
    results = []
    for root, dirs, files in os.walk(target_dir):
        for file in files:
            if file.endswith('.html') or file.endswith('.htm'):
                full_path = os.path.join(root, file)
                results.append(analyze_file(full_path))
    return results

def main():
    if len(sys.argv) < 2:
        print("Kullanım: python seo_folder_scanner.py <klasor_yolu>")
        sys.exit(1)
        
    target_dir = sys.argv[1]
    
    print("\n=== SÜPER SEO MOTORU V3.0 (KAPSAMLI KLASÖR TARAMASI) ===")
    print(f"[+] Hedef Dizin: {os.path.abspath(target_dir)}\n")
    
    try:
        report = scan_directory(target_dir)
        
        if not report:
            print("[!] Belirtilen klasörde HTML dosyası bulunamadı.")
            sys.exit(0)
            
        for res in report:
            print(f"📄 Dosya: {res['file']}")
            
            if res['size'] > 14336:
                print(f"   [X] HATA (TOKEN-BLOAT-001): Boyut {res['size']} byte (14KB aşıldı!)")
            else:
                print(f"   [V] PASS: Boyut {res['size']} byte")
                
            print(f"   [V] PASS: Tek H1 (TECH-H1-001)" if res['h1'] == 1 else f"   [X] HATA: H1 Sayısı {res['h1']}")
            print(f"   [V] PASS: Canonical var (TECH-CANON-001)" if res['canonical'] else f"   [X] HATA: Canonical YOK")
            
            if res['noindex']:
                print(f"   [X] KRİTİK HATA (TECH-NOINDEX-001): Accidental noindex tespit edildi!")
                
            print(f"   [V] PASS: llms.txt discovery" if res['llms_discovery'] else f"   [!] UYARI: describedby=llms.txt YOK")
            print(f"   [V] PASS: data-chunk-id" if res['data_chunk_id'] else f"   [!] UYARI: data-chunk-id bulunamadı")
            print(f"   [V] PASS: H2/H3 Matrisi ({res['h2h3']})" if res['h2h3'] >= 3 else f"   [!] UYARI: H2/H3 yapısı zayıf ({res['h2h3']})")
            print(f"   [V] PASS: Wikidata referansı" if res['wikidata'] else f"   [!] UYARI: Wikidata (QID) eksik")
            
            print("-" * 50)
            
    except Exception as e:
        print(f"[X] HATA: {e}")

if __name__ == "__main__":
    main()
